spoc/io.py

```diff
@@ -73,12 +104,149 @@ def load_chromosome_sizes(path: str):
+        # instantiate pixel parameters
+        pixels = [PixelParameters(**params) for params in metadata.values()]
+        return pixels
+
+    @staticmethod
+    def list_contacts(path: str) -> List[ContactsParameters]:
+        """List available contacts"""
+        # read metadata.json
+        metadata = FileManager._load_metadata(path)
+        # instantiate pixel parameters
+        contacts = [ContactsParameters(**params) for params in metadata.values()]
+        return contacts
+
+    def load_pixels(
+        self, path: str, global_parameters: PixelParameters, load_dataframe: bool = True
+    ) -> Pixels:
+        """Loads specific pixels instance based on global parameters.
+        load_dataframe specifies whether the dataframe should be loaded, or whether pixels
+         should be instantiated based on the path alone.
+
+        Args:
+            path (str): Path to the pixel data.
+            global_parameters (PixelParameters): Global parameters.
+            load_dataframe (bool, optional): Whether to load the dataframe. Defaults to True.
+
+        Returns:
+            Pixels: Pixels object containing the pixel data.
+
+        """
+        metadata = self._load_metadata(path)
+        # find matching pixels
+        for pixel_path, value in metadata.items():
+            param_value = PixelParameters(**value)
+            if param_value == global_parameters:
+                selected_value = param_value
+                break
+        else:
+            raise ValueError(f"No matching pixels found for {global_parameters}")
+        # rewrite path to contain parent folder
+        pixel_path = Path(path) / pixel_path
+        if load_dataframe:
+            df = self._parquet_reader_func(pixel_path)
+        else:
+            df = pixel_path
+        return Pixels(df, **selected_value.dict())
+
+    def load_contacts(
+        self, path: str, global_parameters: ContactsParameters
+    ) -> Contacts:
+        """Loads specific contacts instance based on global parameters.
+        load_dataframe specifies whether the dataframe should be loaded"""
+        metadata = self._load_metadata(path)
+        # find matching contacts
+        for contacts_path, value in metadata.items():
+            param_value = ContactsParameters(**value)
+            if param_value == global_parameters:
+                selected_value = param_value
+                break
+        else:
+            raise ValueError(f"No matching contacts found for {global_parameters}")
+        # rewrite path to contain parent folder
+        contacts_path = Path(path) / contacts_path
+        df = self._parquet_reader_func(contacts_path)
+        return Contacts(df, **selected_value.dict())
 
     @staticmethod
-    def write_pixels(path: str, pixels: Pixels) -> None:
-        """Write pixels"""
-        pixels.data.to_parquet(path, row_group_size=1024*1024)
+    def _get_object_hash_path(path: str, data_object: Union[Pixels, Contacts]) -> str:
```

Comment:
I generally oppose hashing parquet filenames. They are only machine-readable, so it will make impossible for developers to create libraries in other languages that can write Pixels/Contacts, as well as for users to visually inspect the container and maybe open individual parquet files with other software.

Reply:
Discussed in person, hashing will use md5